### PR TITLE
feat(design-sytem): add sliding panel to XDS

### PR DESCRIPTION
**Pull request by diegopf**
EX-7687

### CODE REVIEW

File: packages/x-tailwindcss/.eslintrc.js [hunk 1]
```diff
@@ -2,5 +2,8 @@ module.exports = {
   parserOptions: {
     tsconfigRootDir: __dirname,
     project: 'tsconfig.json'
+  },
+  rules: {
+    '@typescript-eslint/explicit-function-return-type': 'off'
```
Comment (diegopf) [inline packages/x-tailwindcss/.eslintrc.js:7]: We are disabling this rule in almost every file for this package. I think makes sense just disable  it entirely. I will create a PR afterwards removing the `// estlint-disable-next-line ...` if this change goes through.

File: packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/buttons.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/**
+ * Returns the button position variants for the component `sliding panel`.
+ *
+ * @returns The {@link CssStyleOptions} for the variant.
+ */
+export function slidingPanelButtons() {
+  return {
+    '.sliding-panel-button': {
```
Comment (ajperezbau) [inline packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/buttons.ts:8]: Should we split these buttons into different components?
Maybe one for `buttons-center` and another for `buttons-outside`.

**Review by ajperezbau — COMMENTED** [review]
Good job! 👌

File: packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/buttons.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      pointerEvents: 'none',
+      position: 'absolute',
+      transition: 'all ease-out 0.2s',
+      zIndex: 2,
```
Comment (ajperezbau) [inline packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/buttons.ts:12]: Would a `zIndex: 1` be enough here considering that the `sliding-panel` has it set to 0?

File: packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/fade.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * Returns the `fade` variants for the component `sliding panel`.
+ *
+ * @param helpers - The {@link TailwindHelpers} to generate CSS.
+ * @returns The {@link CssStyleOptions} for the variant.
+ */
+export function slidingPanelFade({ theme }: TailwindHelpers) {
+  /* eslint-disable @typescript-eslint/restrict-template-expressions */
+  return {
+    '&.fade': {
+      '.sliding-panel-scroll': {
```
[the remark this answers is not in the document]
Comment (annacv) [reply packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/fade.ts:13]: Agree with this point

File: packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/buttons.ts (hunk 3 above)
Comment (ajperezbau) [inline packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/buttons.ts:12]: Do we need this `zIndex`?

File: packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/fade.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { TailwindHelpers } from '../../../types';
+
+/**
+ * Returns the `fade` variants for the component `sliding panel`.
+ *
+ * @param helpers - The {@link TailwindHelpers} to generate CSS.
+ * @returns The {@link CssStyleOptions} for the variant.
+ */
+export function slidingPanelFade({ theme }: TailwindHelpers) {
+  /* eslint-disable @typescript-eslint/restrict-template-expressions */
```
Comment (ajperezbau) [inline packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/fade.ts:10]: Should we disable this lint for the whole project?

File: packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/buttons.ts (hunk 2 above)
Comment (annacv) [reply packages/x-tailwindcss/src/x-tailwind-plugin/components/sliding-panel/buttons.ts:8]: Hi, I have some doubts here about extracting the hover instead.

**Review by annacv — DISMISSED** [review]
🚀

**Review by ajperezbau — APPROVED** [review]
Good job!

**Review by annacv — APPROVED** [review]
great job!

**Pull request merged**